fiftyone/plugins/utils.py

```diff
@@ -219,3 +222,103 @@ def _do_get_plugin_info(task):
             spec = gh_repo
 
         logger.debug("Failed to retrieve plugin info for %s: %s", spec, e)
+
+
+def get_target_view(ctx, target):
+    """
+    Get the target view for the current context. This is a helper function that
+    returns the view based on the `target` parameter. To be used in conjunction
+    with `list_target_views()`.
+
+    Args:
+        ctx: the current context
+        target: the target view
+
+    Returns:
+        the target view
+
+    """
+    if target == "SELECTED_SAMPLES":
+        return ctx.view.select(ctx.selected)
+
+    if target == "DATASET":
+        return ctx.dataset
+
+    return ctx.view
+
+
+def list_target_views(ctx, inputs):
+    """
+    List the available target views for the current context. This is a helper
+    function that sets the `target` parameter based on the current context. Use
+    it in the `inputs` method of a plugin to allow the user to select the target
+    view from selected samples, the current view, or the entire dataset. To be
+    used in conjunction with `get_target_view()`.
+
+    Args:
+        ctx: the current context
+        inputs: the inputs object
+
+    Returns:
+        None
+
+    """
+    has_view = ctx.view != ctx.dataset.view()
+    has_selected = bool(ctx.selected)
+    default_target = "DATASET"
+    if has_view or has_selected:
+        target_choices = types.RadioGroup()
+        target_choices.add_choice(
+            "DATASET",
+            label="Entire dataset",
+            description="Run on the entire dataset",
```

Comment:
A couple things that prevented me from building a utility like this myself:
1. I found myself preferring slightly different descriptions for each option depending on what operation is being performed
2. I wanted to change descriptions from this UI element to indicate whether the user is processing a dataset or a view
3. I need access to the chosen target view so I can do things like getting schemas on the correct collection (although this particular point could be addressed by just calling `get_target_view()` immediately after `list_target_views()`)

For example, here's a more sophisticated usage of target views that I wrote:
https://github.com/voxel51/fiftyone-plugins/blob/6d1de93dee2cd32d04b1f3071bbbb844697c00c5/plugins/utils/__init__.py#L1525

When considering writing a utility like this, I though to myself: well, if I'm calling a utility but then have to specify which view types to choose between and also the descriptions for each of them, how much effort am I really saving the plugin developer here?

Food for thought 🍪

Reply:
No need to re-invent the wheel. A new built in type and utility would make all of this possible.

```py
# override the defaults
inputs.target_view_selection("target_view", label="Select a View", description="Custom description", other_option="something else")
# or just use the defaults
inputs.target_view_selection("target_view")
```

Reply:
Isn't `inputs.select_view` referenced [here](https://github.com/voxel51/fiftyone/pull/4054#issuecomment-1918056225) actually shorthand for this?

```py
inputs.add_property("target_view", prop)
```

and therefore the way you should access the corresponding parameter would be:

```py
target_view = ctx.params["target_view"]
```

?

Reply:
Here's a similar but different use case:
https://github.com/voxel51/fiftyone-plugins/blob/6d1de93dee2cd32d04b1f3071bbbb844697c00c5/plugins/io/__init__.py#L1069

where the src collection may be the current dataset, current view, or another dataset (or potentially a view into another dataset).

Perhaps that's too niche and doesn't need to be covered by any builtin op here; just pointing out the use case.